aten/src/ATen/native/sparse/SparseTensor.cpp

```diff
@@ -397,13 +399,15 @@ void _validate_sparse_coo_tensor_args(
       "), but got ",
       size.size());
 
-  TORCH_CHECK(
-      indices.is_pinned() == values.is_pinned(),
-      "memory pinning of indices (=",
-      indices.is_pinned(),
-      ") must match memory pinning of values (=",
-      values.is_pinned(),
-      ")");
+  if (check_pinning) {
+    TORCH_CHECK(
```

Comment:
Nit should probably be an assert error more than a RuntimeError, right?
```suggestion
    TORCH_CHECK_ASSERT(

Reply:
@pearu If we are cehcking user inputs, shouldn't we raise a ValueError?

Reply:
No. User inputs can be invalid in many ways, not just with respect to the user provided values. In this particular case, user input is invalid if the memory pinning property of `indices` and `values` is different. This case could be compared to if the devices of `indices` and `values` are different. Within pytorch, such errors are typically exposed as RuntimeError exceptions.